src/databricks/labs/ucx/source_code/linters/python_ast.py

```diff
@@ -0,0 +1,324 @@
+    def args_count(cls, node: Call) -> int:
+        """Count the number of arguments (positionals + keywords)"""
+        assert isinstance(node, Call)
+        return len(node.args) + len(node.keywords)
+
+    @classmethod
+    def get_arg(
+        cls,
+        node: Call,
+        arg_index: int | None,
+        arg_name: str | None,
+    ) -> NodeNG | None:
+        """Extract the call argument identified by an optional position or name (if it has one)"""
+        assert isinstance(node, Call)
+        if arg_index is not None and len(node.args) > arg_index:
+            return node.args[arg_index]
+        if arg_name is not None:
+            arg = [kw.value for kw in node.keywords if kw.arg == arg_name]
+            if len(arg) == 1:
+                return arg[0]
+        return None
+
+    @classmethod
+    def get_full_attribute_name(cls, node: Attribute) -> str:
+        return cls._get_attribute_value(node)
+
+    @classmethod
+    def get_full_function_name(cls, node: Call) -> str | None:
+        if not isinstance(node, Call):
+            return None
+        if isinstance(node.func, Attribute):
+            return cls._get_attribute_value(node.func)
+        if isinstance(node.func, Name):
+            return node.func.name
+        return None
+
+    @classmethod
+    def _get_attribute_value(cls, node: Attribute):
+        if isinstance(node.expr, Name):
+            return node.expr.name + '.' + node.attrname
+        if isinstance(node.expr, Attribute):
+            parent = cls._get_attribute_value(node.expr)
+            return node.attrname if parent is None else parent + '.' + node.attrname
+        if isinstance(node.expr, Call):
+            name = cls.get_full_function_name(node.expr)
+            return node.attrname if name is None else name + '.' + node.attrname
+        name = type(node.expr).__name__
+        if name not in missing_handlers:
+            missing_handlers.add(name)
+            logger.debug(f"Missing handler for {name}")
+        return None
+
+    @classmethod
+    def is_none(cls, node: NodeNG) -> bool:
+        """Check if the given AST expression is the None constant"""
+        if not isinstance(node, Const):
+            return False
+        return node.value is None
+
+    def __repr__(self):
+        truncate_after = 32
+        code = repr(self._root)
+        if len(code) > truncate_after:
+            code = code[0:truncate_after] + "..."
+        return f"<ASTLinter: {code}>"
+
+
+class TreeVisitor:
+
+    def visit(self, node: NodeNG):
+        self._visit_specific(node)
+        for child in node.get_children():
+            self.visit(child)
+
+    def _visit_specific(self, node: NodeNG):
+        method_name = "visit_" + type(node).__name__.lower()
+        method_slot = getattr(self, method_name, None)
+        if callable(method_slot):
+            method_slot(node)
+        else:
+            self.visit_nodeng(node)
+
+    def visit_nodeng(self, node: NodeNG):
+        pass
```

Comment:
```suggestion
        if not callable(method_slot):
            return
        method_slot(node)
```
Simplify

Reply:
We need to call `visit_nodeng` when `visit_xxx` is not callable